server/cache.go

```diff
@@ -470,6 +470,11 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
 				}
 			}
 		}
+		if c.regionStats != nil {
+			for _, item := range overlaps {
+				c.regionStats.clearDefunctRegion(item.GetId())
```

Comment:
Is there some way to call this function in RemoveRegion(), maybe other places will call the RemoveRegion() in the future

Reply:
I think we should use less this method, otherwise, the logic is more confusing.

Reply:
I misunderstood, but now the statistics not belong to `RegionsInfo`, it also needs to know some stores information.